src/components/course/data/utils.jsx

```diff
@@ -870,3 +873,70 @@ export function transformedCourseMetadata({
   };
   return minimalCourseMetadata;
 }
+
+/**
+ * Takes assignments with the earliestPossibleExpirationDate field and sorts by the
+ * soonest expiring expiration date, along with returns relevant metadata
+ *
+ * @param assignmentObjectArray
+ * @param dateFormat
+ * @returns {
+ *  {
+ *    date: (*|string),
+ *    reason: string,
+ *    sortedExpirationDateData: *,
+ *    soonestExpirationDateData: *
+ *   } |
+ *   {
+ *    date: null,
+ *    reason: null,
+ *    sortedExpirationDateData: null
+ *   } |
+ *   {
+ *    date: null,
+ *    reason: null,
+ *    sortedByExpirationDate: null
+ *   }
+ * }
+ */
+export function getSoonestEarliestPossibleExpirationData({
+  assignments,
+  dateFormat = null,
+}) {
+  if (!assignments?.length) {
+    logInfo(`[sortedByExpirationDate] ${assignments} an empty array`);
+    return {
+      soonestExpirationDate: null,
+      soonestExpirationReason: null,
+      soonestExpiringAssignment: null,
+      sortedExpirationAssignments: null,
+    };
+  }
+  const assignmentsWithExpiration = assignments.filter(
+    assignment => !!assignment?.earliestPossibleExpiration,
```

Comment:
nit: is the `?.` needed? I'm assuming we wouldn't have `assignments` array contain values like `undefined`/`null`, etc.?